src/command_modules/azure-cli-webapp/azure/cli/command_modules/webapp/custom.py

```diff
@@ -29,16 +29,29 @@
 
 #workaround that app service's error doesn't comform to LRO spec
 class AppServiceLongRunningOperation(LongRunningOperation): #pylint: disable=too-few-public-methods
+
+    def __init__(self, creating_plan=False):
+        super(AppServiceLongRunningOperation, self).__init__(self)
+        self._creating_plan = creating_plan
+
     def __call__(self, poller):
         try:
             return super(AppServiceLongRunningOperation, self).__call__(poller)
         except Exception as ex:
-            raise AppServiceLongRunningOperation._get_detail_error(ex)
+            raise self._get_detail_error(ex)
 
-    @staticmethod
-    def _get_detail_error(ex):
+    def _get_detail_error(self, ex):
         try:
             detail = json.loads(ex.response.text)['Message']
+            if self._creating_plan:
+                if 'Requested features are not supported in region' in detail:
+                    detail = ("Plan with linux worker is not supported in current region. " +
+                              "Run 'az appservice list-locations --linux-workers-enabled' " +
+                              "to cross check")
+                elif 'Not enough available reserved instance servers to satisfy' in detail:
+                    detail = ("Plan with linux worker can only be created in a group " +
+                              "which has never contained a windows worker. Please use " +
+                              "a new resoruce group. Original error:" + detail)
```

Comment:
This seems like a really weird thing, but in any case, Linux and Windows should be capitalized and there's a type on resource in line 54.